propka/input.py

```diff
@@ -135,11 +135,13 @@ def read_parameter_file(input_file, parameters):
     # try to locate the parameter file
     try:
         ifile = resource_filename(__name__, input_file)
-        input_ = open_file_for_reading(ifile)
+        input_, stream = open_file_for_reading(ifile)
     except (IOError, FileNotFoundError, ValueError, KeyError):
-        input_ = open_file_for_reading(input_file)
+        input_, stream = open_file_for_reading(input_file)
     for line in input_:
         parameters.parse_line(line)
+    if not stream:
```

Comment:
Couldn't this be done more cleanly by either checking for duck-typed attributes or simply using `isinstance`?